### PR TITLE
Predefined property on install

### PR DESCRIPTION
**Pull request by IoannisByng**

### CODE REVIEW

File: lib/Byng/Pimcore/Sitemap/SitemapPlugin.php [hunk 1]
```diff
@@ -16,6 +16,7 @@
 namespace Byng\Pimcore\Sitemap;
 
 use Pimcore\API\Plugin as PluginLib;
+use Pimcore\Model\Property\Predefined;
```
Comment (cj123) [inline lib/Byng/Pimcore/Sitemap/SitemapPlugin.php]: perhaps import this as PredefinedProperty? It'll just make it easier to see what it is, like you did below?

File: README.md [hunk 2]
```diff
@@ -5,16 +5,12 @@ XML sitemap generator for Pimcore. Built from the document tree.
 ## Installation
 1. Run the command ```composer require byng/pimcore-sitemap-plugin```
 2. Enable the plugin from Extras => Extensions on Pimcore admin panel 
+3. Click install to set up ```sitemap_exclude``` property
```
Comment (seeruk) [inline README.md]: Just a heads up on these, the three backticks are generally used for multi-line sections of code, like:

'''
<?php

class Foo extends Bar {
'''

(Replacing the ''' with ``` of course).

For in-line code, `like this here`, you just have to do one backtick either side, like 'hello', but replacing ' with `.

File: lib/Byng/Pimcore/Sitemap/SitemapPlugin.php [hunk 3]
```diff
@@ -36,6 +37,7 @@ public function init()
     {
         parent::init();
 
+
```
Comment (seeruk) [inline lib/Byng/Pimcore/Sitemap/SitemapPlugin.php]: Can we ditch this blank line please?

File: README.md (hunk 2 above)
Comment (IoannisByng) [inline README.md]: will update

File: lib/Byng/Pimcore/Sitemap/SitemapPlugin.php (hunk 1 above)
Comment (IoannisByng) [inline lib/Byng/Pimcore/Sitemap/SitemapPlugin.php]: sure

File: lib/Byng/Pimcore/Sitemap/SitemapPlugin.php (hunk 3 above)
Comment (IoannisByng) [inline lib/Byng/Pimcore/Sitemap/SitemapPlugin.php]: sure

File: lib/Byng/Pimcore/Sitemap/SitemapPlugin.php [hunk 4]
```diff
@@ -52,22 +54,51 @@ public function init()
...
     }
 
     /**
      * {@inheritdoc}
      */
     public static function isInstalled()
     {
-        return true;
+        $property = Predefined::getByKey("sitemap_exclude");
+        if ($property->getId()) {
```
Comment (seeruk) [inline lib/Byng/Pimcore/Sitemap/SitemapPlugin.php]: I'm worried what might happen if `$property` doesn't exist. Could you make the `if` condition the following please:

```
if ($property && $property->getId()) {
```

Comment (IoannisByng) [inline lib/Byng/Pimcore/Sitemap/SitemapPlugin.php]: If it doesn't exist, it returns an instance of Pimcore\Model\Property\Predefined but all its values are null. Which is weird. In the beginning I was checking `if ($property)` and it was never failing. So I am not sure if we actually need to check if the property exists.

Comment (seeruk) [inline lib/Byng/Pimcore/Sitemap/SitemapPlugin.php]: I'm just wondering, if it changes in the future to be just null, it's not gonna hurt to add it, and might save a few headaches later :)

Comment (IoannisByng) [inline lib/Byng/Pimcore/Sitemap/SitemapPlugin.php]: yeap for that reason I would add it :)
I will update the code after the meeting

**Comment by seeruk** [comment]
LGTM, ~~I suspect at some point we may need to deal with child pages, but we can visit that when it comes up.~~ Was thinking of the robots.txt thing we were talking about last week there.

**Pull request merged**